### PR TITLE
Assign multiplayer star rating

### PR DESCRIPTION
**Pull request by timiimit**
- [x] Depends on ppy/osu/pull/23551 & a game package with said PR
- [x] Depends on https://github.com/ppy/osu-server-spectator/pull/174 by necessity

This retrieves beatmap's `difficultyrating` from the db and uses it to populate `MultiplayerPlaylistItem` which is then used by the client to show latest information about room's star rating .

### CODE REVIEW

File: osu.Server.Spectator/Database/DatabaseAccess.cs [hunk 1]
```diff
@@ -67,11 +67,6 @@ public async Task<multiplayer_room> GetRoomAsync(long roomId)
             });
         }
 
-        public async Task<string?> GetBeatmapChecksumAsync(int beatmapId)
-        {
-            return (await GetBeatmapAsync(beatmapId))?.checksum;
-        }
```
Comment (bdach) [inline osu.Server.Spectator/Database/DatabaseAccess.cs:73]: This method is extensively mocked in unit tests, so removing it and not adjusting the tests that mock `GetBeatmapChecksumAsync()` to mock `GetBeatmapAsync()` instead makes the unit tests not compile and also not pass.

Comment (timiimit) [reply osu.Server.Spectator/Database/DatabaseAccess.cs:73]: I replaced all uses of `GetBeatmapChecksumAsync()`.

Comment (bdach) [reply osu.Server.Spectator/Database/DatabaseAccess.cs:73]: You missed one, but I fixed it for you in ed21209d1f8a650c38d76d2c9695e19745a76d0c.

**Comment by bdach** [comment]
Seems fine, but given the fact that this needs a whole lot of things to happen before merge (merge of https://github.com/ppy/osu-server-spectator/pull/174 first, then a game package with https://github.com/ppy/osu/pull/23551, which is going to be in the release-after-next), I'm just going to let this hang unapproved until CI can be actually unblocked.

**Review by bdach — APPROVED** [review]

**Review by peppy — APPROVED** [review]

**Pull request merged**